pkg/reconciler/tenancy/clusterworkspacedeletion/clusterworkspace_deletion_controller.go

```diff
@@ -248,9 +259,23 @@ func (c *Controller) finalizeWorkspace(ctx context.Context, workspace *tenancyv1
 		if workspace.Finalizers[i] == deletion.WorkspaceFinalizer {
 			workspace.Finalizers = append(workspace.Finalizers[:i], workspace.Finalizers[i+1:]...)
 
+			lcn := logicalcluster.From(workspace)
+			listOpts := metav1.ListOptions{
+				LabelSelector: helper.WorkspaceLabelSelector(workspace.Name),
+			}
+
+			// TODO(hasheddan): ClusterRole and ClusterRoleBinding cleanup
+			// should be handled by garbage collection when the controller is
+			// implemented.
+			if err := c.kubeClusterClient.Cluster(lcn).RbacV1().ClusterRoles().DeleteCollection(ctx, backgroudDeletion, listOpts); err != nil && !kerrors.IsNotFound(err) {
```

Comment:
```suggestion
			if err := c.kubeClusterClient.RbacV1().ClusterRoles().DeleteCollection(logicalcluster.WithCluster(ctx, lcn), backgroudDeletion, listOpts); err != nil && !kerrors.IsNotFound(err) {
```

Reply:
no

Reply:
Can you explain how this case is distinct from the existing use of the `kcpClusterClient` ? as shown in line https://github.com/kcp-dev/kcp/pull/2113/files/c1d4b8380218e20ec36ad913ffdcdaa153acb200#diff-13a6c58d33e07a3a3d8bc6d2fdeda5f798da6150f69dde58f7a51b455fc4bc18R277 ?